src/tomato/ketchup/functions.py

```diff
@@ -57,20 +59,27 @@ def submit(args: Namespace) -> None:
     dirs = setlib.get_dirs(args.test)
     settings = setlib.get_settings(dirs.user_config_dir, dirs.user_data_dir)
     queue = settings["queue"]
-    if os.path.exists(args.payload) and os.path.isfile(args.payload):
-        log.debug(f"attempting to open Payload at '{args.payload}'")
-    else:
-        log.error(f"Payload file '{args.payload} not found.")
-        return None
-
-    with open(args.payload, "r") as infile:
-        if args.payload.endswith("json"):
-            pldict = json.load(infile)
-        elif args.payload.endswith("yml") or args.payload.endswith("yaml"):
-            pldict = yaml.full_load(infile)
+    if args.json:  # read directly from json string encoded in base64
+        log.debug("attempting to open payload delivered by base64 encoded json string")
+        encoded_json_string = args.payload
+        json_string = base64.b64decode(encoded_json_string).decode()
+        pldict = json.loads(json_string)
+        payload = to_payload(**pldict)
+    else:  # load a local file
+        if os.path.exists(args.payload) and os.path.isfile(args.payload):
+            log.debug(f"attempting to open Payload at '{args.payload}'")
         else:
-            log.error("Payload file name must end with one of: {json, yml, yaml}.")
+            log.error(f"Payload file '{args.payload} not found.")
             return None
+
+        with open(args.payload, "r") as infile:
+            if args.payload.endswith("json"):
+                pldict = json.load(infile)
+            elif args.payload.endswith("yml") or args.payload.endswith("yaml"):
+                pldict = yaml.full_load(infile)
```

Comment:
Not sure why I'm using `yaml.full_load()` here. Could you check with a `safe_load` and see if the tests pass?

Reply:
I can't get the pytest to work properly so can't check this

9/10 tests fail with assert 'cd' == 'c'

Not sure if it is the cause but I see:

```
INFO:dgbowl_schemas.tomato:Could not parse 'kwargs' using Payload v0.2.
INFO:dgbowl_schemas.tomato:1 validation error for Payload
version
  unexpected value; permitted: '0.2' (type=value_error.const; given=0.1; permitted=('0.2',))
DEBUG:tomato.ketchup.functions:Payload=Payload(version='0.1' tomato=Tomato(unlock_when_done=False, verbosity='WARNING', output=Output(path=None, prefix=None)) sample=Sample(name='dummy_random_5_2') method=[Method(device='worker', technique='random', delay=2, time=5)])
```

I can send you the full log if you want

Reply:
Sorry, the test fails with `safe_load` or fails always with this PR?

Reply:
Always even without this PR, on one PC I get 1 test passing, on another I get all fails